tutorials/xarray_tutorial.py

```diff
@@ -6,295 +6,344 @@
 
-# To verify, we can inspect all their names
-print([coord.name for coord in (x, y, vertical, time)])
+x, y = temperature.metpy.coordinates('x', 'y')
 
 #########################################################################
-# Indexing and Selecting Data
-# ---------------------------
-#
-# MetPy provides wrappers for the usual xarray indexing and selection routines that can handle
-# quantities with units. For DataArrays, MetPy also allows using the coordinate axis types
-# mentioned above as aliases for the coordinates. And so, if we wanted 850 hPa heights,
-# we would take:
+# These coordinate type aliases can also be used in MetPy's wrapped ``.sel`` and ``.loc``
+# for indexing and selecting on ``DataArray``\s. For example, to access 500 hPa heights at
+# 1800Z,
 
-print(data['height'].metpy.sel(vertical=850 * units.hPa))
+heights = data['Geopotential_height_isobaric'].metpy.sel(
+    time='2017-09-05 18:00',
+    vertical=50000.
+)
 
 #########################################################################
-# For full details on xarray indexing/selection, see
-# `xarray's documentation <http://xarray.pydata.org/en/stable/indexing.html>`_.
+# (Notice how we specified 50000 here without units...we'll go over a better alternative in
+# the next section on units.)
+#
+# One point of warning: xarray's selection and indexing only works if these coordinates are
+# *dimension coordinates*, meaning that they are 1D and share the name of their associated
+# dimension. In practice, this means that you can't index a dataset that has 2D latitude and
+# longitude coordinates by latitudes and longitudes, instead, you must index by the 1D y and x
+# dimension coordinates. (What if these coordinates are missing, you may ask? See the final
+# subsection on ``.assign_y_x`` for more details.)
+#
+# Beyond just the coordinates themselves, a common need for both calculations with and plots
+# of geospatial data is knowing the coordinate reference system (CRS) on which the horizontal
+# spatial coordinates are defined. MetPy follows the `CF Conventions
+# <http://cfconventions.org/Data/cf-conventions/cf-conventions-1.8/cf-conventions.html#grid-mappings-and-projections>`_
+# for its CRS definitions, which it then caches on the ``metpy_crs`` coordinate in order for
+# it to persist through calculations and other array operations. There are two ways to do so
+# in MetPy:
+#
+# First, if your dataset is already conforming to the CF Conventions, it will have a grid
+# mapping variable that is associated with the other data variables by the ``grid_mapping``
+# attribute. This is automatically parsed via the ``.parse_cf()`` method:
+
+# Parse full dataset
+data_parsed = data.metpy.parse_cf()
+
+# Parse subset of dataset
+data_subset = data.metpy.parse_cf([
+    'u-component_of_wind_isobaric',
+    'v-component_of_wind_isobaric',
+    'Vertical_velocity_pressure_isobaric'
+])
+
+# Parse single variable
+temperature = data.metpy.parse_cf('Temperature_isobaric')
```

Comment:
LGTM is complaining about this line because the assignment is unused. I'm not sure if there's an easy way around that which makes sense, though.

Reply:
I could choose a different variable from the dataset to demonstrate?

Reply:
Sure.

Reply:
I hate doing things just to make the bots happy. On the other hand, in normal code this is definitely something I want it to flag.